LLama/ChatSession.cs

```diff
@@ -135,30 +135,54 @@ public virtual void LoadSession(string path)
         }
 
         /// <summary>
-        /// Get the response from the LLama model. Note that prompt could not only be the preset words, 
-        /// but also the question you want to ask.
+        /// Generates a response for a given user prompt and manages history state for the user.
+        /// This will always pass the whole history to the model. Don't pass a whole history
+        /// to this method as the user prompt will be appended to the history of the current session.
+        /// If more control is needed, use the other overload of this method that accepts a ChatHistory object.
         /// </summary>
         /// <param name="prompt"></param>
         /// <param name="inferenceParams"></param>
         /// <param name="cancellationToken"></param>
-        /// <returns></returns>
+        /// <returns>Returns generated tokens of the assistant message.</returns>
         public async IAsyncEnumerable<string> ChatAsync(string prompt, IInferenceParams? inferenceParams = null, [EnumeratorCancellation] CancellationToken cancellationToken = default)
         {
-            foreach(var inputTransform in InputTransformPipeline)
+            foreach (var inputTransform in InputTransformPipeline)
                 prompt = inputTransform.Transform(prompt);
-            
-            History.Messages.AddRange(HistoryTransform.TextToHistory(AuthorRole.User, prompt).Messages);
+
+            History.Messages.Add(new ChatHistory.Message(AuthorRole.User, prompt));
+
+            string internalPrompt = HistoryTransform.HistoryToText(History);
```

Comment:
This looks like it prompts with the _entire history_ every time, that doesn't seem correct? Unless you're using a stateless executor then all of the history is still stored within the context KV cache.

Reply:
That was not the intention. I updated the master branch to only use the full history the first time it runs. This enables the use/developer to prime the conversation with a system message and a given back and forth between the user and assistant before handing it off to the first chat message the user enters.